common/src/calendar/util.ts

```diff
@@ -109,17 +109,13 @@ const areOpeningTimesAvailable = (
   slotDate: Date
 ): boolean => {
   return !!openingHours?.some((oh) => {
-    const startDate = oh.date;
-    const startDateTime = new Date(`${startDate}T${oh.startTime}`);
-    const endDateTime = new Date(`${startDate}T${oh.endTime}`);
+    if (oh.state !== "open") return false;
 
-    return (
-      toApiDate(slotDate) === startDate?.toString() &&
-      slotDate < endDateTime &&
-      startDateTime.getDay() === slotDate.getDay() &&
-      startDateTime.getHours() <= slotDate.getHours() &&
-      endDateTime.getHours() >= slotDate.getHours()
-    );
+    const { startTime, endTime } = oh;
+    const startDate = new Date(startTime as string);
+    const endDate = new Date(endTime as string);
```

Comment:
Type cast looks dangerous.
new Date(undefined) => will cause a run time error
explicit null check and return false early so it doesn't need 'as'